main/src/test/java/org/mobilitydata/gtfsvalidator/table/GtfsFareLegRuleDuplicateKeyTest.java

```diff
@@ -0,0 +1,45 @@
+package org.mobilitydata.gtfsvalidator.table;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import org.junit.Test;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.testing.LoadingHelper;
+
+public class GtfsFareLegRuleDuplicateKeyTest {
+  private LoadingHelper helper = new LoadingHelper();
+
+  private static GtfsFareLegRule createFareLegRule(
+      String networkId,
+      String fromAreaId,
+      String toAreaId,
+      String fromTimeframeGroupId,
+      String toTimeframeGroupId,
+      String fareProductId) {
+    return new GtfsFareLegRule.Builder()
+        .setCsvRowNumber(1)
+        .setNetworkId(networkId)
+        .setFromAreaId(fromAreaId)
+        .setToAreaId(toAreaId)
+        .setFromTimeframeGroupId(fromTimeframeGroupId)
+        .setToTimeframeGroupId(toTimeframeGroupId)
+        .setFareProductId(fareProductId)
+        .build();
+  }
+
+  @Test
+  public void noDuplicateKeyNoticeWithNetworkIdAsPrimaryKey() {
```

Comment:
Maybe you could add a test proving that a duplicate key notice is raised if network is the same?